openmc/openmoc_compatible.py

```diff
@@ -0,0 +1,747 @@
+        C = openmc_surface.c
+        D = openmc_surface.d
+        openmoc_surface = openmoc.Plane(A, B, C, D, surface_id, name)
+
+    elif openmc_surface.type == 'x-plane':
+        x0 = openmc_surface.x0
+        openmoc_surface = openmoc.XPlane(x0, surface_id, name)
+
+    elif openmc_surface.type == 'y-plane':
+        y0 = openmc_surface.y0
+        openmoc_surface = openmoc.YPlane(y0, surface_id, name)
+
+    elif openmc_surface.type == 'z-plane':
+        z0 = openmc_surface.z0
+        openmoc_surface = openmoc.ZPlane(z0, surface_id, name)
+
+    elif openmc_surface.type == 'z-cylinder':
+        x0 = openmc_surface.x0
+        y0 = openmc_surface.y0
+        R = openmc_surface.r
+        openmoc_surface = openmoc.ZCylinder(x0, y0, R, surface_id, name)
+
+    else:
+        msg = 'Unable to create an OpenMOC Surface from an OpenMC ' \
+              'Surface of type "{0}" since it is not a compatible ' \
+              'Surface type in OpenMOC'.format(type(openmc_surface))
+        raise ValueError(msg)
+
+    # Set the boundary condition for this Surface
+    openmoc_surface.setBoundaryType(boundary)
+
+    # Add the OpenMC Surface to the global collection of all OpenMC Surfaces
+    OPENMC_SURFACES[surface_id] = openmc_surface
+
+    # Add the OpenMOC Surface to the global collection of all OpenMOC Surfaces
+    OPENMOC_SURFACES[surface_id] = openmoc_surface
+
+    return openmoc_surface
+
+
+def get_openmc_surface(openmoc_surface):
+    """Return an OpenMC surface corresponding to an OpenMOC surface.
+
+    Parameters
+    ----------
+    openmoc_surface : openmoc.Surface
+        OpenMOC surface
+
+    Returns
+    -------
+    openmc_surface : openmc.surface.Surface
+        Equivalent OpenMC surface
+
+    """
+
+    cv.check_type('openmoc_surface', openmoc_surface, openmoc.Surface)
+
+    surface_id = openmoc_surface.id
+
+    # If this Surface was already created, use it
+    if surface_id in OPENMC_SURFACES:
+        return OPENMC_SURFACES[surface_id]
+
+    # Create an OpenMC Surface to represent this OpenCG Surface
+    name = openmoc_surface.name
+
+    # Correct for OpenMC's syntax for Surfaces dividing Cells
+    boundary = openmoc_surface.getBoundaryType()
+    if boundary == openmoc.VACCUM:
+        boundary = 'vacuum'
+    elif boundary == openmoc.REFLECTIVE:
+        boundary = 'reflective'
+    elif boundary == openmoc.PERIODIC:
+        boundary = 'periodic'
+    else:
+        boundary = 'transmission'
+
+    if openmoc_surface.getSurfaceType() == openmoc.PLANE:
+        A = openmoc_surface.getA()
+        B = openmoc_surface.getB()
+        C = openmoc_surface.getC()
+        D = openmoc_surface.getD()
+        openmc_surface = openmc.Plane(surface_id, boundary, A, B, C, D, name)
+
+    elif openmoc_surface.getSurfaceType() == openmoc.XPLANE:
+        x0 = openmoc_surface.getX()
+        openmc_surface = openmc.XPlane(surface_id, boundary, x0, name)
+
+    elif openmoc_surface.type == openmoc.YPLANE:
```

Comment:
Here you use `openmoc_surface.type`, but above you use `openmoc_surface.getSurfaceType()`. Which should it be?